### ISSUE TITLE
Redo dependencies once new Swashbuckle packages are available

### ISSUE THREAD

**Issue by bherila** [bherila/Swashbuckle.AspNetCore.AzureFunctions#2]

**Comment by jhoerr** [comment]
Is this issue related to the `BH.Swashbuckle.AspNetCore.*` package references in the `Swashbuckle.AspNetCore.AzureFunctions.csproj`? Can you elaborate which SwashBuckle package updates you're waiting on? 

I'm stumbling when I attempt to use the `Swashbuckle.AspNetCore.AzureFunctions` along with another community project, `Swashbuckle.AspNetCore.Filters`. When I add the latest `Filters` package, I get compilation errors in the `AddSwaggerGen` action. Alternately, If I change the `BH.Swashbuckle.*` to their latest `Swashbuckle.*` counterparts, the compilation errors go away, but the function throws this error when generating the OpenAPI document:

```
System.Private.CoreLib: Exception while executing function: OpenAPI. 
functions: The type initializer for '<StartupCode$functions>.$Functions' threw an exception.
System.Private.CoreLib: The given key 'controller' was not present in the dictionary. 
```

I'm wondering if that `controller` key is something being added in your `BH.Swashbuckle.*` code?

**Comment by bherila** [comment]
These dependencies should work:

```
    <PackageReference Include="Swashbuckle.AspNetCore.AzureFunctions" Version="1.1.1" />    
    <PackageReference Include="System.ComponentModel.Annotations" Version="4.5.0" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="2.1.4" />
    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="1.0.24" />
```

I need to find some time to remove the dependencies on the BH packages, and test to make sure everything is working.

**Comment by bherila** [comment]
Also, my proposed edits to Swashbuckle (in the BH packages) have already been merged to the master branch in the main Swashbuckle Core repos. I am not sure if they have released new packages since my PR.

**Comment by jhoerr** [comment]
5.0.0-beta was published on 11/24. It appears to include all your accepted PRs (913,914,940). 4.0.1

**Comment by jhoerr** [comment]
When I update your project and my function app to reference `Swashbuckle.AspNetCore.Swagger 5.0.0-beta`, the response schema disappears. I *think* what maybe happened is the `5.0.0-beta` package is derived from [this PR](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/pull/952), which was branched from master before your PRs were merged into master.

I'll try cloning and referencing `Swashbuckle.AspNetCore.Swagger` and `Swashbuckle.AspNetCore.Annotations` directly from the `master` branch to see if the problem goes away.

**Comment by bherila** [comment]
Thanks for investigating this, I also tried to rebase my branch to master but got a couple of merge conflicts along the way. 

I would love to get this all unified again but have been really pressed for time lately. I will definitely help to get this fixed and update the packages when I can.

**Comment by jhoerr** [comment]
Just to follow up, I updated my function app to directly reference the `master` branch of `Swashbuckle.AspNetCore.*`, and I updated your project to do the same. The response schemas still don't appear. `SwaggerResponse` annotations are still being processed, so there's something else going on. 🤷‍♂️

**Comment by WillEastbury** [comment]
FYI I'm seeing the same trying to update to netcore3.1. In SwaggerGeneratorOptions.cs 

        private IList<string> DefaultTagsSelector(ApiDescription apiDescription)
        {
            return new[] {apiDescription.ActionDescriptor.RouteValues["controller"] };
        }

Throws a keynotfound exception as there is nothing populated in RouteValues at all.

**Comment by kraihn** [comment]
Hey @bherila, are you planning on revisiting this issue and updating for netcore 3.1?

**Comment by WillEastbury** [comment]
I don't think it is a set of trivial changes and there dependencies in swashbuckle too.

I ended up rewriting my own package (Buccaneer) here, this might help you in the meantime generating OpenApi specs. 

https://github.com/WillEastbury/Buccaneer

**Comment by MisinformedDNA** [comment]
I filed a PR, but did not receive a response. In the meantime, I've created an [updated package](https://github.com/MisinformedDNA/Swashbuckle.AspNetCore.AzureFunctions/packages?q=M) on GitHub Package Registry.
